lib/execution_engine2/utils/arg_processing.py

```diff
@@ -33,4 +38,124 @@ def parse_bool(putative_bool: Union[str, bool, int, float, None]) -> bool:
         if pb.lower() == "false":
             return False
 
-    raise ValueError(f"{pb} is not a boolean value")
+    raise IncorrectParamsException(f"{pb} is not a boolean value")
+
+
+# The remaining methods are ported from
+# https://github.com/kbase/sample_service/blob/master/lib/SampleService/core/arg_checkers.py
+# with slight changes.
+# Should probably make a package or see if there are equivalent 3rd party functions at some point.
+# Although if you want to use custom exceptions as here that won't work
+
+
+def not_falsy(item: T, item_name: str) -> T:
+    """
+    Check if a value is falsy and throw and exception if so.
+    :param item: the item to check for falsiness.
+    :param item_name: the name of the item to include in any exception.
+    :raises ValueError: if the item is falsy.
+    :returns: the item.
+    """
+    if not item:
+        raise ValueError(f"{item_name} cannot be a value that evaluates to false")
+    return item
+
+
+def not_falsy_in_iterable(
+    iterable: Optional[Iterable[T]], name: str, allow_none: bool = False
+) -> Optional[Iterable[T]]:
+    """
+    Check that an iterable is not None and contains no falsy items. Empty iterables are accepted.
+    :param iterable: the iterable to check.
+    :param name: the name of the iterable to be used in error messages.
+    :param allow_none: allow the iterable to be None - in this case return None. The contents of
+        the iterable may not be None.
+    :returns: the iterable.
+    :raises ValueError: if the iterable is None or contains falsy items.
+    """
+    # probably need to allow for 0 as an option
+    if iterable is None:
+        if allow_none:
+            return None
+        raise ValueError(f"{name} cannot be None")
+    for i, item in enumerate(iterable):
+        not_falsy(item, f"Index {i} of iterable {name}")
+    return iterable
+
+
+def _contains_control_characters(string: str) -> bool:
+    """
+    Check if a string contains control characters, as denoted by the Unicode character category
+    starting with a C.
+    :param string: the string to check.
+    :returns: True if the string contains control characters, False otherwise.
+    """
+    # make public if needed
+    # See https://stackoverflow.com/questions/4324790/removing-control-characters-from-a-string-in-python  # noqa: E501
+    for c in string:
+        if unicodedata.category(c)[0] == "C":
+            return True
+    return False
+
+
+def _no_control_characters(string: str, name: str) -> str:
```

Comment:
Does this need to be in a separate method from `_contains_control_characters`? Are there situations where you are likely to use the latter but not the former?

Reply:
Way back when I wrote these originally, I had them split up because I did use them separately. However, that situation went away and so I made them private. Hence the docs on internal methods, which I usually don't do.

I could imagine a situation where you might want to scan a set of strings for control characters and report to the user which ones are bad rather than throwing an exception on the first one, or something along those lines, so there's at least a theoretical use case where splitting the method makes sense.

The functions are easy enough to grok that I don't think having them split up adds significant cognitive burden, and since they're not part of the API they don't add testing burden, so my preference is to leave things as is. I don't feel particularly strongly about it though.